llvm/lib/Target/AMDGPU/Utils/AMDGPUDelayedMCExpr.h

```diff
@@ -0,0 +1,39 @@
+//===- AMDGPUDelayedMCExpr.h - Delayed MCExpr resolve -----------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_LIB_TARGET_AMDGPU_UTILS_AMDGPUDELAYEDMCEXPR_H
+#define LLVM_LIB_TARGET_AMDGPU_UTILS_AMDGPUDELAYEDMCEXPR_H
+
+#include "llvm/BinaryFormat/MsgPackDocument.h"
+#include <deque>
+
+namespace llvm {
+class MCExpr;
+
+class DelayedMCExpr {
```

Comment:
Should this type name be plural? And is there significance to the fact that the outer type includes "MC" in the name but not the inner? Is it just to make the name shorter?

With the namespacing from nesting I'd say it could be even shorter, maybe:

```
class DelayedMCExprs {
    struct Expr {...}
}
```

Reply:
Duplicated this comment in parent PR https://github.com/llvm/llvm-project/pull/93236